spec/spec_helper.rb

```diff
@@ -1 +1,9 @@
 require 'puppetlabs_spec_helper/module_spec_helper'
+require 'rspec-puppet'
```

Comment:
Strange thing, it works for me with both Puppet 4.3 and 3.8 (fresh system running in Docker). I think you have some other Rspec settings interfering. Are you running the test with `bundle exec rake spec`?